File: Harvest/Views/Components/ChipView.swift
import SwiftUI

struct ChipView: View {
    let title: String
    var isSelected: Bool = false
    var lightStyle: Bool = false
    var onTap: (() -> Void)?

    var body: some View {
        if lightStyle {
            lightChip
        } else {
            Button {
                onTap?()
            } label: {
                Text(title)
            }
            .buttonStyle(.harvestGlass(.chip(selected: isSelected)))
        }
    }

    // Light chips live on cream/white form surfaces, so they keep the
    // solid-capsule treatment for contrast rather than translucent glass.
    private var lightChip: some View {
        Button {
            onTap?()
        } label: {
            Text(title)
                .font(HarvestTheme.Typography.bodySmall)
                .fontWeight(isSelected ? .semibold : .regular)
                .foregroundStyle(isSelected ? HarvestTheme.Colors.textOnRedPrimary : HarvestTheme.Colors.textPrimary)
                .padding(.horizontal, HarvestTheme.Spacing.md)
                .padding(.vertical, HarvestTheme.Spacing.sm)
                .background {
                    if isSelected {
                        Capsule().fill(HarvestTheme.Colors.formAccent)
                    } else {
                        Capsule()
                            .fill(HarvestTheme.Colors.formSurface)
                            .overlay {
                                Capsule().stroke(HarvestTheme.Colors.formBorder, lineWidth: 1)
                            }
                    }
                }
        }
        .buttonStyle(.plain)
    }
}

File: android/app/src/main/java/com/harvestglass/harvest/ui/components/ChipView.kt
package com.harvestglass.harvest.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.harvestglass.harvest.ui.theme.HarvestButton
import com.harvestglass.harvest.ui.theme.HarvestButtonKind
import com.harvestglass.harvest.ui.theme.HarvestTheme

/** Port of Harvest/Views/Components/ChipView.swift. */
@Composable
fun ChipView(
    title: String,
    isSelected: Boolean = false,
    lightStyle: Boolean = false,
    onTap: (() -> Unit)? = null
) {
    if (!lightStyle) {
        HarvestButton(
            text = title,
            kind = if (isSelected) HarvestButtonKind.CHIP_SELECTED else HarvestButtonKind.CHIP_UNSELECTED,
            onClick = { onTap?.invoke() }
        )
        return
    }

    // Light chips live on cream/white form surfaces, so they keep the
    // solid-capsule treatment for contrast rather than translucent glass.
    val shape = RoundedCornerShape(percent = 50)
    Text(
        text = title,
        style = HarvestTheme.Typography.bodySmall,
        fontWeight = if (isSelected) FontWeight.SemiBold else FontWeight.Normal,
        color = if (isSelected) HarvestTheme.Colors.textOnRedPrimary else HarvestTheme.Colors.textPrimary,
        modifier = Modifier
            .background(
                if (isSelected) HarvestTheme.Colors.formAccent else HarvestTheme.Colors.formSurface,
                shape
            )
            .then(
                if (!isSelected) Modifier.border(1.dp, HarvestTheme.Colors.formBorder, shape)
                else Modifier
            )
            .clickable { onTap?.invoke() }
            .padding(horizontal = HarvestTheme.Spacing.md, vertical = HarvestTheme.Spacing.sm)
    )
}
